### ISSUE TITLE
Metricbeat "io.stat: input does not match format"

### ISSUE THREAD

**Issue by matutter** [elastic/beats#32345]
I've been deploying `metricbeat` to my hosts for about a month - this one error persists on every host. I'm not sure what effect this is having on the metricsets I'm gather because I've never had metricbeat run without this error.

**Version**: 8.3.1 (same in 8.3.0, 8.2)
**Operating System:** Ubuntu 22.04 (server & desktop), as well as PopOS
**Discuss Forum URL**: (no reply) https://discuss.elastic.co/t/metricbeat-error-about-scanning-empty-io-stat-file/308063
**Steps to Reproduce**:

1. Deploy `metricbeat:8.3.1` with `docker-compose up`.
2. See error shown in logs.

Log message from docker container:

```
error getting cgroup stats for V2: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: input does not match format
```

Full `stderr` from docker container:

```
{
  "log.level": "error",
  "@timestamp": "2022-07-13T13:00:13.362Z",
  "log.logger": "metrics",
  "log.origin": {
    "file.name": "report/report.go",
    "file.line": 234
  },
  "message": "error getting cgroup stats for V2: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: input does not match format",
  "service.name": "metricbeat",
  "ecs.version": "1.6.0"
}
```

Contents of `/hostfs/sys/fs/cgroup/io.stat`:

**io.stat from Ubuntu 22.04**

Obtained from inside the container using: `docker-compose exec metricbeat cat /hostfs/sys/fs/cgroup/io.stat`

```
253:0 rbytes=3766756352 wbytes=447383982080 rios=124096 wios=60014994 dbytes=1938477453312 dios=18688
259:0 rbytes=3868738560 wbytes=447829653504 rios=119422 wios=29182222 dbytes=1941498814464 dios=18722
7:7 7:6 7:5 7:4 rbytes=556032 wbytes=0 rios=78 wios=0 dbytes=0 dios=0
7:3 rbytes=21017600 wbytes=0 rios=629 wios=0 dbytes=0 dios=0
7:2 rbytes=2377728 wbytes=0 rios=153 wios=0 dbytes=0 dios=0
7:1 rbytes=2580480 wbytes=0 rios=148 wios=0 dbytes=0 dios=0
7:0 rbytes=3140608 wbytes=0 rios=360 wios=0 dbytes=0 dios=0
```

**io.stat from PopOS**

```
253:2 rbytes=3006474240 wbytes=5773680640 rios=733028 wios=1409598 dbytes=0 dios=0
253:1 rbytes=45929317376 wbytes=211814596608 rios=1078342 wios=21388321 dbytes=0 dios=0
253:0 rbytes=45931053056 wbytes=211814596608 rios=1078394 wios=21426614 dbytes=0 dios=0
259:0 rbytes=48963873792 wbytes=217588278272 rios=1315370 wios=15358572 dbytes=3222265856 dios=24
7:7 7:6 7:5 7:4 7:3 7:2 7:1 7:0
```

Full `metricbeat.yml`:

```yaml
logging.to_stderr: true
logging.level: error

output.logstash:
  hosts: ["logstash.test.lab:5044"]

monitoring.enabled: false

metricbeat.autodiscover:
  providers:
    - type: docker
      hints.enabled: true

metricbeat.modules:

  # metrics that are collected more frequently
  - module: system
    hostfs: /hostfs
    period: 10s
    cpu.metrics:  [percentages]
    core.metrics: [percentages]
    metricsets:
      - cpu      # no priv
      - load     # no priv
      - memory   # no priv, e.g. swap, total, used, free, actual
      - network  # no priv, network metrics for interfaces

  # metrics that are collected less frequently
  - module: system
    hostfs: /hostfs
    period: 1m
    processors:
      - drop_event.when.regexp:
          system.filesystem.mount_point: '^/(sys|cgroup|proc|dev|etc|host|lib)($|/)'
    metricsets:
      - filesystem  # no priv, e.g. size, free, used

  # metrics that are collected rarely
    hostfs: /hostfs
  - module: system
    period: 15m
    metricsets:
      - uptime      # no priv

  - module: docker
    metricsets:
      - cpu
      - diskio
      - event
      - info
      - memory
      - network
    hosts:
      - unix:///var/run/docker.sock
    period: 10s

  - module: docker
    period: 1m
    hosts:
      - unix:///var/run/docker.sock
    metricsets:
      - healthcheck

  - module: docker
    period: 15m
    hosts:
      - unix:///var/run/docker.sock
    metricsets:
      - container

processors:

  - drop_fields:
      ignore_missing: true
      fields:
        - docker.container.labels
        - docker.event.actor.attributes

  - add_host_metadata:
      netinfo.enabled: true
```

Docker compose file `docker-compose.yml`:

```yaml
version: "3.9"

networks:
  beats_net:
    internal: false

services:

  metricbeat:
    image: docker.elastic.co/beats/metricbeat:8.3.1

    hostname: 'test.lab'

    user: root

    privileged: true

    volumes:
      - /home/bitworker/test/sim2/beats/metricbeat.yml:/usr/share/metricbeat/metricbeat.yml:ro
      - /proc:/hostfs/proc:ro
      - /:/hostfs:ro
      - /sys/fs/cgroup:/hostfs/sys/fs/cgroup:ro
      - /var/run/docker.sock:/var/run/docker.sock:ro

    restart: unless-stopped

    networks:
      - beats_net
```

**Comment by Nacoma** [comment]
Also seeing this with 

```yml
  elastic-agent:
    image: docker.elastic.co/beats/elastic-agent:8.5.1
    container_name: elastic-agent
    hostname: elastic-agent
    restart: unless-stopped
    user: root
    environment:
      FLEET_ENROLLMENT_TOKEN: ${FLEET_ENROLLMENT_TOKEN}
      FLEET_URL: ${FLEET_URL}
      FLEET_ENROLL: 1
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /proc:/hostfs/proc:ro
      - /sys/fs/cgroup:/hostfs/sys/fs/cgroup:ro
      - /:/hostfs:ro
```

**Comment by sziladriana** [comment]
Having the same problem. We have alerts set up for any error that occurs, this one pollutes the alerts constantly unless it's explicitly ignored. I see the discussion was closed with no replies :(

**Comment by litronics** [comment]
Having the same issue with Raspberry Pi4 running on Debian V 11.6.

The format of my io.stat:
`cat /sys/fs/cgroup/io.stat
179:0 rbytes=1710142464 wbytes=2054013952 rios=36047 wios=191775 dbytes=0 dios=0
7:7 7:6 7:5 7:4 7:3 7:2 7:1 7:0 1:15 1:14 1:13 1:12 1:11 1:10 1:9 1:8 1:7 1:6 1:5 1:4 1:3 1:2 1:1 1:0 `

**Comment by iamjosh007** [comment]
@cmacknz   @Danouchka @elastic/elastic-agent

This is happening on 8.5.3 stack with agents based out of 8.5v..

We tried removing or forced unrolling of all agents running on K8s and removed all agent specific resources deployed as part of daemonset and tried setting up the agent again but still same issue.. 

Logs are flowing on very low phase with most of the service logs are not appearing and metrics (state metrics) associated to kube-state are not showing up but just pod, container .. etc metrics.

Right now ES Dev and support got no idea with our case and its been more than 30 hours (P1)

Integrations enabled prior - K8s, System.

- Agent logs -

{"log.level":"info","@timestamp":"2023-02-22T19:31:08.803Z","log.origin":{"file.name":"stateresolver/stateresolver.go","file.line":66},"message":"Updating internal state","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2023-02-22T19:31:11.296Z","log.origin":{"file.name":"report/report.go","file.line":234},"message":"error getting cgroup stats for V2: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: input does not match format","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2023-02-22T19:31:21.297Z","log.origin":{"file.name":"report/report.go","file.line":234},"message":"error getting cgroup stats for V2: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: input does not match format","ecs.version":"1.6.0"}
{"log.level":"error","@timestamp":"2023-02-22T19:31:31.300Z","log.origin":{"file.name":"report/report.go","file.line":234},"message":"error getting cgroup stats for V2: error fetching stats for controller io: error fetching IO stats: error getting io.stats for path /hostfs/sys/fs/cgroup: error scanning file: /hostfs/sys/fs/cgroup/io.stat: input does not match format","ecs.version":"1.6.0"}

**Comment by cmacknz** [comment]
The cgroup format is tied to the Linux kernel version, it is possible a hardware/OS/kernel upgrade triggered this error which will require a Metricbeat update to fix.

It would surprise me if this error had consequences other than failing to collect cgroup related Metrics.

**Comment by Nacoma** [comment]
The additional consequences we're facing are

- containers are polluting the infrastructure inventory because they're considered hosts
- host machine infrastructure is not associated with traces

**Comment by mgaruccio** [comment]
I'm also seeing a bunch of this error, if it's helpful the nodes are on kernel 5.15.60 and ubuntu 22.04

**Comment by laraMorenoIgle** [comment]
This issue is fixed in [elastic/elastic-agent-system-metrics#77 ](https://github.com/elastic/elastic-agent-system-metrics/pull/77) that is included in `8.7.1` (not in `8.7.0`  as tagged)
